routes/projects.rb

```diff
@@ -70,13 +70,16 @@ def self.registered(app)
         project.update(name:  params[:name] || project.name,
                        description: params[:description] || project.description,
                        active: params[:active] || project.active)
+
         redirect "/projects/#{params[:id]}"
       end
 
       app.delete '/projects/:id/?' do
         # delete a project
         project = Project[id: params[:id]]
-        project.delete unless project.nil?
+        # disassociate this projects' resources to the default project and
+        # delete this project
+        project.reassign_resources(project) unless project.name == 'default'
```

Comment:
In addition, we should probably not show a delete button on the default client/project, it'll be confusing that it is there but doesn't do anything.